packages/contracts/contracts/DependencyRegistryV0.sol

```diff
@@ -727,10 +776,31 @@ contract DependencyRegistryV0 is
             dependencyDetails[_dependencyType].scriptBytecodeAddresses[_index];
     }
 
+    /**
+     * @notice Returns the storage library version for
+     *         dependency `_dependencyTypes` at script index `_index`.
+     */
+    function getDependencyScriptBytecodeStorageVersionAtIndex(
+        bytes32 _dependencyType,
+        uint256 _index
+    ) external view returns (bytes32) {
+        return
+            BytecodeStorageReader.getLibraryVersionForBytecode(
+                dependencyDetails[_dependencyType].scriptBytecodeAddresses[
+                    _index
```

Comment:
nit: might be worth checking if this is the zero address and return with intuitive error message?

Reply:
Rather than checking if this is the zero address, I think its more appropriate to apply the check on index out of bounds querying here, above, and more holistically throughout this PR which I've applied in the latest commit. 

For example, I've also updated this behavior below, to fail in this manner rather than optimistically returning the empty string, which I'm not sure is behavior that makes sense (I think we should fail more clearly when attempting to read out of bounds content).

WDYT about this approach @ryley-o @yoshiwarab ?

Reply:
ah, yes this makes sense to me 👍

Reply:
Vibey – TY ser!